cve_bin_tool/checkers/zlib.py

```diff
@@ -17,9 +17,15 @@ class ZlibChecker(Checker):
     CONTAINS_PATTERNS = [
         rf"deflate[{printable}]*Copyright 1995-2005 Jean-loup Gailly",
         rf"Copyright 1995-2005 Mark Adler[{printable}]*inflate",
+        r"Copyright 1995-2017 Jean-loup Gailly and Mark Adler",
+        r"Copyright 1995-2017 Mark Adler",
         r"too many length or distance symbols",
     ]
-    FILENAME_PATTERNS = [r"libz.so."]
+    FILENAME_PATTERNS = [
+        r"libz.so.",
+        r"libz.so.[01]",
+        r"libz.so.([01]+\.[0-9]+\.[0-9])",
```

Comment:
Those last two patterns don't contain anything that won't also be matched by the first pattern, so they are redundant and should be removed.

If you're not sure how these regexes work, you can always play around with them in the console:

```
$ python
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AM
D64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> re.search(r"libz.so.", "libz.so.1")
<re.Match object; span=(0, 8), match='libz.so.'>
>>> re.search(r"libz.so.", "no match here")
>>>
```